tests/worker/test_workflow.py

```diff
@@ -0,0 +1,989 @@
+            await assert_eq_eventually("signal 2", last_event)
+
+        # Confirm two logs happened
+        assert find_log("Signal: signal 1 ({'workflow_id':")
+        assert find_log("Signal: signal 2")
+        assert not find_log("Signal: signal 3")
+        # Also make sure it has some workflow info
+        record = find_log("Signal: signal 1")
+        assert (
+            record
+            and record.__dict__["workflow_info"].workflow_type == "LoggingWorkflow"
+        )
+
+        # Clear queue and start a new one with more signals
+        log_queue.queue.clear()
+        async with new_worker(
+            client, LoggingWorkflow, task_queue=worker.task_queue
+        ) as worker:
+            # Send a couple signals
+            await handle.signal(LoggingWorkflow.my_signal, "signal 3")
+            await handle.signal(LoggingWorkflow.my_signal, "finish")
+            await handle.result()
+
+        # Confirm replayed logs are not present but new ones are
+        assert not find_log("Signal: signal 1")
+        assert not find_log("Signal: signal 2")
+        assert find_log("Signal: signal 3")
+        assert find_log("Signal: finish")
+    finally:
+        workflow.logger.base_logger.removeHandler(handler)
+        workflow.logger.base_logger.setLevel(prev_level)
+
+
+@workflow.defn
+class StackTraceWorkflow:
+    def __init__(self) -> None:
+        self._status = "created"
+
+    @workflow.run
+    async def run(self) -> None:
+        # Start several tasks
+        awaitables = [
+            asyncio.sleep(1000),
+            workflow.execute_activity(
+                wait_cancel, schedule_to_close_timeout=timedelta(seconds=1000)
+            ),
+            workflow.execute_child_workflow(
+                LongSleepWorkflow.run, id=f"{workflow.info().workflow_id}_child"
+            ),
+            self.never_completing_coroutine(),
+        ]
+        await asyncio.wait([asyncio.create_task(v) for v in awaitables])
+
+    async def never_completing_coroutine(self) -> None:
+        self._status = "waiting"
+        await workflow.wait_condition(lambda: False)
+
+    @workflow.query
+    def status(self) -> str:
+        return self._status
+
+
+async def test_workflow_stack_trace(client: Client):
+    async with new_worker(
+        client, StackTraceWorkflow, LongSleepWorkflow, activities=[wait_cancel]
+    ) as worker:
+        handle = await client.start_workflow(
+            StackTraceWorkflow.run,
+            id=f"workflow-{uuid.uuid4()}",
+            task_queue=worker.task_queue,
+        )
+        # Wait until waiting
+        async def status() -> str:
+            return await handle.query(StackTraceWorkflow.status)
+
+        await assert_eq_eventually("waiting", status)
+
+        # Send stack trace query
+        trace = await handle.query("__stack_trace")
+        # TODO(cretz): Do more specific checks once we clean up traces
+        assert "never_completing_coroutine" in trace
+
+
+# TODO:
```

Comment:
There's probably more, I'd review the other SDKs' tests to see if there's anything important we're missing.